src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/UseBed.java

```diff
@@ -0,0 +1,56 @@
+package protocolsupport.protocol.packet.middleimpl.clientbound.play.v_pe;
+
+import protocolsupport.protocol.ConnectionImpl;
+import protocolsupport.protocol.packet.middle.clientbound.play.MiddleUseBed;
+import protocolsupport.protocol.packet.middleimpl.ClientBoundPacketData;
+import protocolsupport.protocol.serializer.DataWatcherSerializer;
+import protocolsupport.protocol.serializer.PositionSerializer;
+import protocolsupport.protocol.serializer.VarNumberSerializer;
+import protocolsupport.protocol.typeremapper.pe.PEPacketIDs;
+import protocolsupport.protocol.utils.datawatcher.DataWatcherObject;
+import protocolsupport.protocol.utils.datawatcher.objects.DataWatcherObjectByte;
+import protocolsupport.protocol.utils.datawatcher.objects.DataWatcherObjectVector3vi;
+import protocolsupport.protocol.utils.networkentity.NetworkEntity;
+import protocolsupport.utils.CollectionsUtils;
+import protocolsupport.utils.recyclable.RecyclableArrayList;
+import protocolsupport.utils.recyclable.RecyclableCollection;
+
+public class UseBed extends MiddleUseBed {
+
+	public UseBed(ConnectionImpl connection) {
+		super(connection);
+	}
+
+	private static final int DATA_PLAYER_BED_POSITION = 28;
+	private static final int TYPE_PLAYER_SPAWN = 0;
+	private static final byte DATA_PLAYER_FLAG_SLEEP = 2;  // Nukkit has this as 1 << 1
+
+	private ClientBoundPacketData createSingleAttributeDataPacket(NetworkEntity player, int key, DataWatcherObject<?> value) {
```

Comment:
All of that has to be moved and merged with existing entity metadata handling logic.

Reply:
Ok, done. The existing logic needed some tender love and care as well.